File: practice/src/main/java/com/decade/practice/web/rest/OnlineController.java
package com.decade.practice.web.rest;

import com.decade.practice.model.OnlineStatus;
import com.decade.practice.presence.UserPresenceService;
import org.springframework.web.bind.annotation.GetMapping;
import org.springframework.web.bind.annotation.PathVariable;
import org.springframework.web.bind.annotation.RequestMapping;
import org.springframework.web.bind.annotation.RestController;

import java.security.Principal;
import java.util.List;

@RestController
@RequestMapping("/onlines")
public class OnlineController {

        private final UserPresenceService stat;

        public OnlineController(UserPresenceService stat) {
                this.stat = stat;
        }

        @GetMapping
        public List<OnlineStatus> listOnline(Principal principal) {
                return stat.getOnlineList(principal.getName());
        }

        @GetMapping("/{username}")
        public OnlineStatus get(@PathVariable String username) {
                return stat.get(username);
        }
}

File: practice/src/main/java/com/decade/practice/controllers/OnlineController.kt
package com.decade.practice.controllers

import com.decade.practice.core.OnlineStatistic
import com.fasterxml.jackson.databind.ObjectMapper
import com.fasterxml.jackson.databind.SerializationFeature
import org.springframework.web.bind.annotation.GetMapping
import org.springframework.web.bind.annotation.PathVariable
import org.springframework.web.bind.annotation.RequestMapping
import org.springframework.web.bind.annotation.RestController
import java.security.Principal

@RestController
@RequestMapping("/online")
class OnlineController(private val stat: OnlineStatistic) {

      @GetMapping
      fun listOnline(principal: Principal) = stat.getOnlineList(principal.name).also {
            println(ObjectMapper().enable(SerializationFeature.INDENT_OUTPUT).writeValueAsString(it))
      }

      @GetMapping("/{username}")
      fun get(@PathVariable username: String) = stat.get(username)


}